crates/trie/sparse/src/trie.rs

```diff
@@ -579,29 +579,29 @@ impl RevealedSparseTrie {
     /// Returns a list of paths to the nodes that are located at the provided depth when counting
     /// from the root node. If there's a leaf at a depth less than the provided depth, it will be
     /// included in the result.
-    fn get_nodes_at_depth(&self, depth: usize) -> HashSet<Nibbles> {
+    fn get_nodes_at_depth(&self, depth: usize) -> Vec<Nibbles> {
         let mut paths = Vec::from([(Nibbles::default(), 0)]);
-        let mut targets = HashSet::<Nibbles>::default();
+        let mut targets = Vec::new();
```

Comment:
vec with capacity or smallvec?

Reply:
we cannot estimate how many targets will be collected. It depends on the trie structure, so we can't just do `16^N` to get the number of nodes at level `N`.